src/midi/midi_lasynth_model.cpp

```diff
@@ -50,48 +51,72 @@ LASynthModel::LASynthModel(const std::string &rom_name,
 	assert(ctrl_full || (ctrl_a && ctrl_b));
 }
 
+// Scans the requested directory for the requested ROM.
+// Uses a static cache of directories and their ROMs to avoid repeat scans.
 std::optional<std_fs::path> LASynthModel::find_rom(const Mt32ServicePtr& service,
                                                    const std_fs::path& dir,
                                                    const Rom* rom)
 {
-	static std::unordered_set<std::string> unknown_files;
 	if (!rom) {
-		return {};
+		return std::nullopt;
 	}
 
-	std::error_code ec;
+	// Named types to make the directory cache self-documenting
+	using RomId     = std::string;
+	using RomPath   = std_fs::path;
+	using DirPath   = std_fs::path;
+	using RomsCache = std::map<RomId, RomPath>;
+	using DirsCache = std::map<DirPath, RomsCache>;
+	using JunkFiles = std::set<RomPath>;
+
+	static DirsCache dirs_cache = {};
+	static JunkFiles junk_files = {};
```

Comment:
Just `static std::set<std_fs::path> unknown_files = {}`?

Reply:
Yup; at first I created  std::map<std_fs::path, std::map<std::string, std_fs::path> > and thought "nope, this looks like a maze" so just aliased each key value.. but dropping the `std_fs::path`s is a better balance and is still self-documenting enough. 

Fixed; thanks @johnnovak!

Reply:
Yeah, some type aliasing can help, but I'm finding too much just obscures the intent. I generally steer away from aliasing simple types directly, but `RomId = std::string` helps here for sure.

I find quite often you need to turn these types into simple structs anyway if you need method-like behaviour on them. E.g., check out my `struct ImageInfo` thing in `video.h`.

Reply:
Thanks for the tip; will check it out next time 👍